src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs

```diff
@@ -1,221 +1,371 @@
+                        break;
+                }
+            }
+        }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.AnyUnix)]
-    public void UnixFilePathWithSpaces()
-    {
-        Assert.Throws<ArgumentException>(() => GetEntries("\0"));
-        using (File.Create(Path.Combine(TestDirectory, " ")))
-        using (File.Create(Path.Combine(TestDirectory, "          ")))
-        using (File.Create(Path.Combine(TestDirectory, "\n")))
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public void UnixSearchPatternInvalid()
         {
-            String[] results = GetEntries(TestDirectory);
-            Assert.Contains(Path.Combine(TestDirectory, " "), results);
-            Assert.Contains(Path.Combine(TestDirectory, "          "), results);
-            Assert.Contains(Path.Combine(TestDirectory, "\n"), results);
+            Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, "\0"));
+            Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, string.Format("te{0}st", "\0".ToString())));
         }
-    }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.AnyUnix)]
-    public void UnixSearchPatternWithSpaces()
-    {
-        Assert.Empty(GetEntries(TestDirectory, "           "));
-        Assert.Empty(GetEntries(TestDirectory, "\n"));
-        Assert.Empty(GetEntries(TestDirectory, " "));
-        Assert.Empty(GetEntries(TestDirectory, ""));
-        Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, "\0"));
-    }
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsSearchPatternWildCharacters()
+        {
+            string strTempDir = Path.Combine("dls;d", "442349-0", "v443094(*)(+*$#$*") + new string(Path.DirectorySeparatorChar, 3);
+            Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, strTempDir));
+        }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.AnyUnix)]
-    public void UnixSearchPatternWithDoubleDots()
-    {
-        // search pattern is valid but directory doesn't exist
-        Assert.Throws<DirectoryNotFoundException>(() => GetEntries(TestDirectory, Path.Combine("..ab ab.. .. abc..d", "abc..")));
-
-        // invalid search pattern trying to go up a directory with ..
-        Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, Path.Combine("..ab ab.. .. abc..d", "abc", "..")));
-        Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, Path.Combine("..ab ab.. .. abc..d", "..", "abc")));
-        Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, Path.Combine("..", "..ab ab.. .. abc..d", "abc")));
-        Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, Path.Combine("..", "..ab ab.. .. abc..d", "abc") + Path.DirectorySeparatorChar));
-    }
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsSearchPatternQuestionMarks()
```

Comment:
Failed the build system too. I thought I'd removed all the deeply nested test directories, but I appear to have missed this one. Thanks.